src/ngraph_builder.cc

```diff
@@ -1055,49 +964,22 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
       ng::Shape ng_image_shape(2);
       ng::Shape ng_kernel_shape(2);
 
-      if (is_nhwc) {
-        auto& s = ng_input->get_shape();
-        ng::Shape reshaped_shape{s[0], s[3], s[1], s[2]};
-
-        NGRAPH_VLOG(3) << "reshaped_shape: " << ng::join(reshaped_shape);
-
-        ng_input = make_shared<ng::op::Reshape>(
-            ng_input, ng::AxisVector{0, 3, 1, 2}, reshaped_shape);
-
-        ng_strides[0] = tf_strides[1];
-        ng_strides[1] = tf_strides[2];
-
-        ng_dilations[0] = tf_dilations[0];
-        ng_dilations[1] = tf_dilations[1];
-
-        ng_image_shape[0] = s[1];
-        ng_image_shape[1] = s[2];
-      } else {
-        auto& s = ng_input->get_shape();
-
-        ng_strides[0] = tf_strides[1];
-        ng_strides[1] = tf_strides[2];
-
-        ng_dilations[0] = tf_dilations[0];
-        ng_dilations[1] = tf_dilations[1];
-
-        ng_image_shape[0] = s[2];
-        ng_image_shape[1] = s[3];
-      }
+      BatchToNGraph(is_nhwc, ng_input);
+      BatchedOpParamToNGraph(is_nhwc, ng_input->get_shape(), ng_image_shape);
+      // TODO is the below wrong?
```

Comment:
Can someone comment on this?

Reply:
It doesn't look right to me. I believe the pattern should be the same as for the shape.